src/m3ninx/index/segment/fst/encoding/docs/slice.go

```diff
@@ -67,9 +67,8 @@ func (r *SliceReader) Doc(id postings.ID) (doc.Document, error) {
 	return r.Read(id)
 }
 
-// AllDocs returns a docs iterator.
+// Iter returns a docs iterator.
 func (r *SliceReader) Iter() index.IDDocIterator {
-	postingsIter := postings.NewRangeIterator(r.offset,
-		r.offset+postings.ID(r.Len()))
+	postingsIter := postings.NewRangeIterator(0, postings.ID(r.Len()))
```

Comment:
Can't comment on relevant line unfortunately but we can remove offset from `NewSliceReader` and the `SliceReader` struct too